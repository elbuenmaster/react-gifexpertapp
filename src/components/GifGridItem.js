import React from 'react'
import PropTypes from 'prop-types'
export const GifGridItem = ({id,title,url}) => {
    console.log(id)
    return (
        <div className="card animate__fadeIn">
        
           <img src={url} alt={title}></img> 
           <p>{title}</p>       
       
        </div>
    )
}

GifGridItem.propTypes={
    id:PropTypes.number.isRequired,
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,
}
export default GifGridItem;