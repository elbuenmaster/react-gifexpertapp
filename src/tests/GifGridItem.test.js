import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import  GifGridItem  from '../components/GifGridItem';

const id=1;
const title="One Punch";
const url = "https://media4.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=d469d910a8hyirxzp6u6pshnn3owbwv66xbwzg8l8kwsbxw6&amp;rid=giphy.gif"
const wrapper = shallow(<GifGridItem id={id} title={title} url={url}></GifGridItem>);

describe('gridItem',()=>{

    test('testing snapshot',()=>{
       expect(wrapper).toMatchSnapshot();
    });
    test('debe contener un parrafo con el titulo',()=>{
         const parrafo = wrapper.find('p').text();
         expect(parrafo).toBe(title);
    })
    test('propiedades de IMG deben coincidir con las props',()=>{
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.prop('alt')).toBe(title);
   })
   test('debe contener clase animate__fadeIn',()=>{

        const div=wrapper.find('div');
        expect(div.prop('className')).toContain('animate__fadeIn');
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
   })
});

