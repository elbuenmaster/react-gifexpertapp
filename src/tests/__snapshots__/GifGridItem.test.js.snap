// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gridItem testing snapshot 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="One Punch"
    src="https://media4.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=d469d910a8hyirxzp6u6pshnn3owbwv66xbwzg8l8kwsbxw6&amp;rid=giphy.gif"
  />
  <p>
    One Punch
  </p>
</div>
`;
